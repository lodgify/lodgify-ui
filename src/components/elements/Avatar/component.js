import React from 'react';
import { string } from 'prop-types';

import { Thumbnail } from 'media/Thumbnail';
import { testidFactory } from 'utils/testid';

const testid = testidFactory('avatar');

export const Component = ({ firstname, lastname, image }) => (
  <div className="avatar" data-testid={testid()}>
    {!!image ? (
      <Thumbnail data-testid={testid('image')} imageUrl={image} isCircular />
    ) : (
      <p data-testid={testid('text')}>
        {`${firstname[0]}${lastname[0]}`.toUpperCase()}
      </p>
    )}
  </div>
);

Component.displayName = 'Avatar';
Component.defaultProps = {
  firstname: ' ',
  lastname: ' ',
  image: null,
};
Component.propTypes = {
  firstname: string,
  image: string,
  lastname: string,
};
