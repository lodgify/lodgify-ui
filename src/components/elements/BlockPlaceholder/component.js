import React from 'react';
import { PropTypes } from 'prop-types';
import classnames from 'classnames';

/**
 * A BlockPlaceholder indicates that a block of content will soon appear in a layout.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ isFluid, isRectangular, isSquare }) => (
  <div
    className={classnames('placeholder block', {
      fluid: isFluid,
      rectangle: isRectangular,
      square: isSquare,
    })}
  />
);

Component.displayName = 'BlockPlaceholder';

Component.defaultProps = {
  isFluid: false,
  isSquare: false,
  isRectangular: false,
};

Component.propTypes = {
  /** The placeholder fills the height and width of its parent. */
  isFluid: PropTypes.bool,
  /** The placeholder maintains 4:3 proportions. */
  isRectangular: PropTypes.bool,
  /** The placeholder maintains 1:1 proportions. */
  isSquare: PropTypes.bool,
};
