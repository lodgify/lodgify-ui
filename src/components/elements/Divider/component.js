import React from 'react';
import PropTypes from 'prop-types';
import { Divider } from 'semantic-ui-react';
import getClassNames from 'classnames';

import { getIsSizeLarge } from './utils/getIsSizeLarge';

/**
 * A divider adds whitespace between other elements.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ className, hasLine, size }) => (
  <Divider
    className={getClassNames(className, {
      'is-size-small': size === 'small',
      'is-size-huge': size === 'huge',
    })}
    hidden={!hasLine}
    section={getIsSizeLarge(size)}
  />
);

Component.displayName = 'Divider';

Component.defaultProps = {
  className: '',
  hasLine: false,
  size: 'medium',
};

Component.propTypes = {
  /**
   * Custom class name.
   * Provided by `ShowOn` so ignored in the styleguide.
   * @ignore
   */
  className: PropTypes.string,
  /** Does the divider have a visible line. */
  hasLine: PropTypes.bool,
  /** The size of the divider. */
  size: PropTypes.oneOf(['small', 'medium', 'large', 'huge']),
};
