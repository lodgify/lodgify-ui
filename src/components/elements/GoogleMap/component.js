import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Card } from 'semantic-ui-react';
import { debounce } from 'debounce';

import { ReactGoogleMap } from 'utils/react-google-maps';

import { getParentNodeWidth } from './utils/getParentNodeWidth';
import { getHeight } from './utils/getHeight';
import { getMaxWidth } from './utils/getMaxWidth';
import { getImgSrc } from './utils/getImgSrc';
import { GOOGLE_MAPS_API_KEY } from './constants';

/**
 * A map displays a location.
 * By default, the map will display a static image then load a dynamic interface on user interaction.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export class Component extends PureComponent {
  state = {
    isDynamicMapShowing: false,
    isMouseOverMap: false,
    parentNodeWidth: 0,
  };

  componentDidMount = () => {
    if (!this.img) return;

    global.addEventListener('resize', this.setParentNodeWidth);
    this.img.addEventListener('mousedown', this.setIsDynamicMapShowingTrue);
    this.img.addEventListener('mouseover', this.setIsMouseOverMapTrue);
    this.img.addEventListener('mouseout', this.setIsMouseOverMapFalse);
    this.setParentNodeWidth();
  };

  componentDidUpdate = (previousProps, previousState) =>
    !this.state.isDynamicMapShowing &&
    !previousState.isMouseOverMap &&
    this.state.isMouseOverMap &&
    global.setTimeout(
      () => this.state.isMouseOverMap && this.setIsDynamicMapShowingTrue(),
      500
    );

  componentWillUnmount = () => {
    if (!this.img) return;

    global.removeEventListener('resize', this.setParentNodeWidth);
    this.img.removeEventListener('mousedown', this.setIsDynamicMapShowingTrue);
    this.img.removeEventListener('mouseover', this.setIsMouseOverMapTrue);
    this.img.removeEventListener('mouseout', this.setIsMouseOverMapFalse);
  };

  createRef = img => (this.img = img);

  setParentNodeWidth = debounce(() => {
    const parentNodeWidth = getParentNodeWidth(this.img);

    this.setState({ parentNodeWidth });
  }, 10);

  setIsDynamicMapShowingTrue = () =>
    this.setState({ isDynamicMapShowing: true });

  setIsMouseOverMapTrue = () => this.setIsMouseOverMap(true);

  setIsMouseOverMapFalse = () => this.setIsMouseOverMap(false);

  setIsMouseOverMap = isMouseOverMap => this.setState({ isMouseOverMap });

  render = () => {
    const {
      apiKey,
      isFullBleed,
      isFluid,
      isShowingExactLocation,
      isShowingApproximateLocation,
      latitude,
      longitude,
      markers,
    } = this.props;

    const height = getHeight(this.props.height, isFullBleed, isFluid);

    return isFullBleed ? (
      <ReactGoogleMap
        apiKey={apiKey}
        height={height}
        isShowingApproximateLocation={isShowingApproximateLocation}
        isShowingExactLocation={isShowingExactLocation}
        latitude={latitude}
        longitude={longitude}
        markers={markers}
      />
    ) : (
      <Card fluid style={{ height, maxWidth: getMaxWidth(isFluid) }}>
        {isFluid || this.state.isDynamicMapShowing ? (
          <ReactGoogleMap
            apiKey={apiKey}
            height={height}
            isShowingApproximateLocation={isShowingApproximateLocation}
            isShowingExactLocation={isShowingExactLocation}
            latitude={latitude}
            longitude={longitude}
            markers={markers}
          />
        ) : (
          <img
            ref={this.createRef}
            src={getImgSrc(
              apiKey,
              height,
              isShowingApproximateLocation,
              isShowingExactLocation,
              latitude,
              longitude,
              this.state.parentNodeWidth
            )}
          />
        )}
      </Card>
    );
  };
}

Component.displayName = 'GoogleMap';

Component.defaultProps = {
  apiKey: GOOGLE_MAPS_API_KEY,
  height: '400px',
  isFluid: false,
  isFullBleed: false,
  isShowingExactLocation: false,
  isShowingApproximateLocation: false,
  latitude: 0,
  longitude: 0,
  markers: [],
};

Component.propTypes = {
  /** An [API key](https://developers.google.com/maps/documentation/javascript/get-api-key) for using Google Maps. */
  apiKey: PropTypes.string,
  /** A valid CSS value to set the height of the map. */
  height: PropTypes.string,
  /** Does the map show on a card which fills the width and height of its container. */
  isFluid: PropTypes.bool,
  /** Does the map fill the width and height of its container. */
  isFullBleed: PropTypes.bool,
  /** Is the map showing a marker for the approximate location at the center of the map. */
  isShowingApproximateLocation: PropTypes.bool,
  /** Is the map showing a marker for the exact location location at the center of the map. */
  isShowingExactLocation: PropTypes.bool,
  /** The latitude coordinate for the center of the map. */
  latitude: PropTypes.number,
  /** The longitude coordinate for the center of the map. */
  longitude: PropTypes.number,
  /** The markers to display on the map. */
  markers: PropTypes.arrayOf(
    PropTypes.shape({
      /** The React component to show as a marker. */
      component: PropTypes.node.isRequired,
      /**
       * A function called to determine the offset of the marker relative to its top left corner.
       * @param   {number} markerPixelWidth
       * @param   {number} markerPixelHeight
       * @return  {object}
       */
      getOffset: PropTypes.func,
      /** The latitude coordinate for the marker. */
      latitude: PropTypes.number,
      /** The longitude coordinate for the marker. */
      longitude: PropTypes.number,
    })
  ),
};
