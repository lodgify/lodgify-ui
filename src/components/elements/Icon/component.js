import React from 'react';
import PropTypes from 'prop-types';
import getClassNames from 'classnames';

import { Paragraph } from 'typography/Paragraph';

import { ICON_NAMES_LIST } from './constants';
import { getPath } from './utils/getPath';

/**
 * An icon is a glyph used to represent something else.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  color,
  hasBorder,
  isCircular,
  isColorInverted,
  isDisabled,
  isLabelLeft,
  labelText,
  labelWeight,
  name,
  path,
  size,
  ...otherProps
}) => (
  <i
    className={getClassNames('icon', color, size, {
      circular: isCircular,
      'has-border': hasBorder,
      'inverted grey': isDisabled,
      'has-label': !!labelText,
      inverted: isColorInverted,
    })}
    // Passing through props is required for
    // compatibility with dynamic components e.g. `ToolTip`
    {...otherProps}
  >
    {!!labelText && isLabelLeft && (
      <Paragraph weight={labelWeight}>{labelText}</Paragraph>
    )}
    <svg viewBox="0 0 24 24">
      <path d={getPath(name, path)} fill="currentColor" />
    </svg>
    {!!labelText && !isLabelLeft && (
      <Paragraph weight={labelWeight}>{labelText}</Paragraph>
    )}
  </i>
);

Component.displayName = 'Icon';

Component.defaultProps = {
  color: null,
  hasBorder: false,
  isCircular: false,
  isColorInverted: false,
  isDisabled: false,
  isLabelLeft: false,
  labelText: null,
  labelWeight: null,
  name: null,
  path: null,
  size: null,
};

Component.propTypes = {
  /** The color of the icon. */
  color: PropTypes.oneOf([
    'red',
    'orange',
    'yellow',
    'olive',
    'green',
    'teal',
    'blue',
    'violet',
    'purple',
    'pink',
    'brown',
    'grey',
    'light grey',
    'black',
  ]),
  /** Is the icon formatted to have a border  */
  hasBorder: PropTypes.bool,
  /** Is the icon formatted to appear circular  */
  isCircular: PropTypes.bool,
  /** Is the color of the icon inverted for contrast */
  isColorInverted: PropTypes.bool,
  /** Is the icon disabled. */
  isDisabled: PropTypes.bool,
  /** Is the label on the left hand side of the icon */
  isLabelLeft: PropTypes.bool,
  /** A visible label to display with the icon. */
  labelText: PropTypes.string,
  /** The weight of the label. */
  labelWeight: PropTypes.oneOf(['heavy', 'light']),
  /**
   * The name of the icon to display. Takes priority over `props.path`.
   * [See here for the full list of valid icon names](https://github.com/lodgify/lodgify-ui/blob/production/src/components/elements/Icon/constants.js)
   */
  name: PropTypes.oneOf(ICON_NAMES_LIST),
  /**
   * A string containing a series of path descriptions. Should fit a 24x24 viewBox.
   * [See MDN for documentation.](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d)
   */
  path: PropTypes.string,
  /** The size of the icon. */
  size: PropTypes.oneOf([
    'mini',
    'tiny',
    'small',
    'large',
    'big',
    'huge',
    'massive',
  ]),
};
