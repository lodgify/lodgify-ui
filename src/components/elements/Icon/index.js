export { Component as Icon } from './component';
export { ICON_NAMES } from './constants';
