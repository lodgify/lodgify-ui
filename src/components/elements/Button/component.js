import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Button } from 'semantic-ui-react';

import { Icon } from 'elements/Icon';

/**
 * A button indicates a possible user action.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  children,
  isDisabled,
  isLoading,
  isPositionedRight,
  onClick,
  isRounded,
  isSecondary,
  isCompact,
  hasShadow,
  icon,
  size,
}) => (
  <Button
    circular={isRounded}
    className={cx({
      'has-shadow': !!hasShadow,
    })}
    compact={isCompact}
    disabled={isDisabled}
    floated={isPositionedRight ? 'right' : 'left'}
    loading={isLoading}
    onClick={onClick}
    secondary={isSecondary}
    size={size}
  >
    {!!icon && <Icon name={icon} />}
    {children}
  </Button>
);

Component.displayName = 'Button';

Component.defaultProps = {
  children: null,
  onClick: Function.prototype,
  isDisabled: false,
  isPositionedRight: false,
  isLoading: false,
  isRounded: false,
  isSecondary: false,
  isCompact: false,
  hasShadow: false,
  icon: null,
  size: null,
};

Component.propTypes = {
  /** The content to display in the button. */
  children: PropTypes.node,
  /** Has the button shadow. */
  hasShadow: PropTypes.bool,
  /** An icon to display in the button
   * [See here for the full list of valid icon names](https://github.com/lodgify/lodgify-ui/blob/master/src/components/elements/Icon/constants.js)
   */
  icon: PropTypes.string,
  /** Has the button got reduced padding. */
  isCompact: PropTypes.bool,
  /** Is the button disabled. */
  isDisabled: PropTypes.bool,
  /** Is the button in loading state. */
  isLoading: PropTypes.bool,
  /** Is the button positioned on the right hand side of its container. */
  isPositionedRight: PropTypes.bool,
  /** Has the button rounded borders. */
  isRounded: PropTypes.bool,
  /** Is the button secondary. */
  isSecondary: PropTypes.bool,
  /** The function to call when the button is clicked.
   *  @param {Object} event
   */
  onClick: PropTypes.func,
  /** The size of the button */
  size: PropTypes.string,
};
