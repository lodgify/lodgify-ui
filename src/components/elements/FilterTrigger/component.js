import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Icon } from 'elements/Icon';
import { MORE_FILTERS } from 'utils/default-strings';

/**
 * A filter trigger removes filters and displays the available filters.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  onFiltersClick,
  onClearFiltersClick,
  filterTriggerText,
  hasFiltersSelected,
}) => (
  <div
    className={classNames('filter-trigger', {
      'has-filter': !!hasFiltersSelected,
    })}
  >
    <button onClick={onFiltersClick}>
      <Icon labelText={filterTriggerText} name="filter" />
    </button>
    {hasFiltersSelected && (
      <button onClick={onClearFiltersClick}>
        <Icon name="close" />
      </button>
    )}
  </div>
);

Component.displayName = 'FilterTrigger';

Component.defaultProps = {
  filterTriggerText: MORE_FILTERS,
  hasFiltersSelected: false,
  onClearFiltersClick: Function.prototype,
  onFiltersClick: Function.prototype,
};

Component.propTypes = {
  /** The text inside the filter trigger. */
  filterTriggerText: PropTypes.string,
  /** Filters are selected. */
  hasFiltersSelected: PropTypes.bool,
  /** The function to call when the clear filters button is clicked.
   *  @param {Object} event
   */
  onClearFiltersClick: PropTypes.func,
  /** The function to call when the filters button is clicked.
   *  @param {Object} event
   */
  onFiltersClick: PropTypes.func,
};
