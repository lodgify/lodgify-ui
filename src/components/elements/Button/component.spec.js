import React from 'react';
import { shallow } from 'enzyme';
import { Button as SemanticButton } from 'semantic-ui-react';
import {
  expectComponentToBe,
  expectComponentToHaveChildren,
  expectComponentToHaveProps,
  expectComponentToHaveDisplayName,
} from '@lodgify/enzyme-jest-expect-helpers';

import { Icon } from 'elements/Icon';

import { Component as Button } from './component';

const getButton = props => shallow(<Button {...props}>Press me</Button>);

describe('<Button />', () => {
  it('should render a single Semantic UI `Button` component', () => {
    const wrapper = getButton();

    expectComponentToBe(wrapper, SemanticButton);
  });

  it('should pass the `Button` component the right props', () => {
    const wrapper = getButton();

    expectComponentToHaveProps(wrapper, {
      disabled: false,
      floated: 'left',
      loading: false,
      circular: false,
      compact: false,
      type: 'button',
    });
  });

  it('should pass the `Button` component `props.isRounded` as `circular`', () => {
    const wrapper = getButton({ isRounded: true });

    expectComponentToHaveProps(wrapper, {
      circular: true,
    });
  });

  describe('if `props.isFormSubmit` is true', () => {
    it('should pass the `Button` component `floated="right"`', () => {
      const wrapper = getButton({ isFormSubmit: true });

      expectComponentToHaveProps(wrapper, {
        type: 'submit',
      });
    });
  });

  describe('if `props.isPositionedRight` is true', () => {
    it('should pass the `Button` component `floated="right"`', () => {
      const wrapper = getButton({ isPositionedRight: true });

      expectComponentToHaveProps(wrapper, {
        floated: 'right',
      });
    });
  });

  describe('if `props.isSecondary` is true', () => {
    it('should pass the `Button` component `secondary={true}`', () => {
      const wrapper = getButton({ isSecondary: true });

      expectComponentToHaveProps(wrapper, {
        secondary: true,
      });
    });
  });

  describe('if `props.isCompact` is true', () => {
    it('should pass the `Button` component `compact={true}`', () => {
      const wrapper = getButton({ isCompact: true });

      expectComponentToHaveProps(wrapper, {
        compact: true,
      });
    });
  });

  describe('if `props.size` is informed', () => {
    it('should pass the `Button` component `size={size}`', () => {
      const wrapper = getButton({ size: 'massive' });

      expectComponentToHaveProps(wrapper, {
        size: 'massive',
      });
    });
  });

  it('should render the right children', () => {
    const wrapper = getButton();

    expectComponentToHaveChildren(wrapper, 'Press me');
  });

  describe('if `props.icon` is informed', () => {
    const getButtonWithIcon = () => getButton({ icon: 'search' });

    it('should render an <Icon />', () => {
      const wrapper = getButtonWithIcon();

      expectComponentToHaveChildren(wrapper, Icon, 'Press me');
    });

    describe('the <Icon>', () => {
      it('should have the right props', () => {
        const wrapper = getButtonWithIcon().find(Icon);

        expectComponentToHaveProps(wrapper, {
          name: 'search',
        });
      });
    });
  });

  describe('if `props.hasShadow` is informed', () => {
    it('should have the `has-shadow` className', () => {
      const wrapper = getButton({ hasShadow: true });

      expectComponentToHaveProps(wrapper, {
        className: 'has-shadow',
      });
    });
  });

  it('should have displayName `Button`', () => {
    expectComponentToHaveDisplayName(Button, 'Button');
  });
});
