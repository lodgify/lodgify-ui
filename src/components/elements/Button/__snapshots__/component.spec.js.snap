// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> if \`props.hasOutline\` is informed should render the right structure 1`] = `
<Button
  hasShadow={false}
  icon={null}
  isCompact={false}
  isDisabled={false}
  isFluid={false}
  isFormSubmit={false}
  isLoading={false}
  isOutlined={true}
  isPositionedRight={false}
  isRounded={false}
  isSecondary={false}
  onClick={[Function]}
  size={null}
>
  <Button
    as="button"
    basic={true}
    circular={false}
    className="has-outline"
    compact={false}
    disabled={false}
    floated="left"
    fluid={false}
    loading={false}
    onClick={[Function]}
    secondary={false}
    size={null}
    type="button"
  >
    <button
      className="ui basic left floated button has-outline"
      onClick={[Function]}
      role="button"
      type="button"
    >
      Press me
    </button>
  </Button>
</Button>
`;

exports[`<Button /> if \`props.hasShadow\` is informed should render the right structure 1`] = `
<Button
  hasShadow={true}
  icon={null}
  isCompact={false}
  isDisabled={false}
  isFluid={false}
  isFormSubmit={false}
  isLoading={false}
  isOutlined={false}
  isPositionedRight={false}
  isRounded={false}
  isSecondary={false}
  onClick={[Function]}
  size={null}
>
  <Button
    as="button"
    basic={false}
    circular={false}
    className="has-shadow"
    compact={false}
    disabled={false}
    floated="left"
    fluid={false}
    loading={false}
    onClick={[Function]}
    secondary={false}
    size={null}
    type="button"
  >
    <button
      className="ui left floated button has-shadow"
      onClick={[Function]}
      role="button"
      type="button"
    >
      Press me
    </button>
  </Button>
</Button>
`;

exports[`<Button /> if \`props.icon\` is informed should render the right structure 1`] = `
<Button
  hasShadow={false}
  icon="search"
  isCompact={false}
  isDisabled={false}
  isFluid={false}
  isFormSubmit={false}
  isLoading={false}
  isOutlined={false}
  isPositionedRight={false}
  isRounded={false}
  isSecondary={false}
  onClick={[Function]}
  size={null}
>
  <Button
    as="button"
    basic={false}
    circular={false}
    className=""
    compact={false}
    disabled={false}
    floated="left"
    fluid={false}
    loading={false}
    onClick={[Function]}
    secondary={false}
    size={null}
    type="button"
  >
    <button
      className="ui left floated button"
      onClick={[Function]}
      role="button"
      type="button"
    >
      <Icon
        color={null}
        hasBorder={false}
        isButton={false}
        isCircular={false}
        isColorInverted={false}
        isDisabled={false}
        isLabelLeft={false}
        labelText={null}
        labelWeight={null}
        name="search"
        path={null}
        size={null}
      >
        <i
          className="icon"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M20.22,4.14a6.38,6.38,0,0,0-9.75,8.2l-1.8,1.74a1.77,1.77,0,0,0-2,.31l-4.14,4a1.77,1.77,0,0,0,0,2.49l.37.39a1.77,1.77,0,0,0,2.5,0l4.15-4a1.79,1.79,0,0,0,.38-2l1.8-1.73a6.39,6.39,0,0,0,8.54-9.46ZM8.69,16.48l-4.15,4a.6.6,0,0,1-.85,0l-.37-.38a.62.62,0,0,1,0-.86l4.15-4a.61.61,0,0,1,.84,0l.2.2.19.21A.61.61,0,0,1,8.69,16.48Zm2.13-3.67h0L11,13Zm.41.41L11,13l.2.18Zm8-.88a5.23,5.23,0,1,1,.13-7.4A5.24,5.24,0,0,1,19.25,12.34Z"
              fill="currentColor"
            />
          </svg>
        </i>
      </Icon>
      Press me
    </button>
  </Button>
</Button>
`;

exports[`<Button /> if \`props.isCompact\` is true should render the right structure 1`] = `
<Button
  hasShadow={false}
  icon={null}
  isCompact={true}
  isDisabled={false}
  isFluid={false}
  isFormSubmit={false}
  isLoading={false}
  isOutlined={false}
  isPositionedRight={false}
  isRounded={false}
  isSecondary={false}
  onClick={[Function]}
  size={null}
>
  <Button
    as="button"
    basic={false}
    circular={false}
    className=""
    compact={true}
    disabled={false}
    floated="left"
    fluid={false}
    loading={false}
    onClick={[Function]}
    secondary={false}
    size={null}
    type="button"
  >
    <button
      className="ui compact left floated button"
      onClick={[Function]}
      role="button"
      type="button"
    >
      Press me
    </button>
  </Button>
</Button>
`;

exports[`<Button /> if \`props.isFormSubmit\` is true should render the right structure 1`] = `
<Button
  hasShadow={false}
  icon={null}
  isCompact={false}
  isDisabled={false}
  isFluid={false}
  isFormSubmit={true}
  isLoading={false}
  isOutlined={false}
  isPositionedRight={false}
  isRounded={false}
  isSecondary={false}
  onClick={[Function]}
  size={null}
>
  <Button
    as="button"
    basic={false}
    circular={false}
    className=""
    compact={false}
    disabled={false}
    floated="left"
    fluid={false}
    loading={false}
    onClick={[Function]}
    secondary={false}
    size={null}
    type="submit"
  >
    <button
      className="ui left floated button"
      onClick={[Function]}
      role="button"
      type="submit"
    >
      Press me
    </button>
  </Button>
</Button>
`;

exports[`<Button /> if \`props.isPositionedRight\` is true should render the right structure 1`] = `
<Button
  hasShadow={false}
  icon={null}
  isCompact={false}
  isDisabled={false}
  isFluid={false}
  isFormSubmit={false}
  isLoading={false}
  isOutlined={false}
  isPositionedRight={true}
  isRounded={false}
  isSecondary={false}
  onClick={[Function]}
  size={null}
>
  <Button
    as="button"
    basic={false}
    circular={false}
    className=""
    compact={false}
    disabled={false}
    floated="right"
    fluid={false}
    loading={false}
    onClick={[Function]}
    secondary={false}
    size={null}
    type="button"
  >
    <button
      className="ui right floated button"
      onClick={[Function]}
      role="button"
      type="button"
    >
      Press me
    </button>
  </Button>
</Button>
`;

exports[`<Button /> if \`props.isRounded\` is true should render the right structure 1`] = `
<Button
  hasShadow={false}
  icon={null}
  isCompact={false}
  isDisabled={false}
  isFluid={false}
  isFormSubmit={false}
  isLoading={false}
  isOutlined={false}
  isPositionedRight={false}
  isRounded={true}
  isSecondary={false}
  onClick={[Function]}
  size={null}
>
  <Button
    as="button"
    basic={false}
    circular={true}
    className=""
    compact={false}
    disabled={false}
    floated="left"
    fluid={false}
    loading={false}
    onClick={[Function]}
    secondary={false}
    size={null}
    type="button"
  >
    <button
      className="ui circular left floated button"
      onClick={[Function]}
      role="button"
      type="button"
    >
      Press me
    </button>
  </Button>
</Button>
`;

exports[`<Button /> if \`props.isSecondary\` is true should render the right structure 1`] = `
<Button
  hasShadow={false}
  icon={null}
  isCompact={false}
  isDisabled={false}
  isFluid={false}
  isFormSubmit={false}
  isLoading={false}
  isOutlined={false}
  isPositionedRight={false}
  isRounded={false}
  isSecondary={true}
  onClick={[Function]}
  size={null}
>
  <Button
    as="button"
    basic={false}
    circular={false}
    className=""
    compact={false}
    disabled={false}
    floated="left"
    fluid={false}
    loading={false}
    onClick={[Function]}
    secondary={true}
    size={null}
    type="button"
  >
    <button
      className="ui secondary left floated button"
      onClick={[Function]}
      role="button"
      type="button"
    >
      Press me
    </button>
  </Button>
</Button>
`;

exports[`<Button /> if \`props.size\` is informed should pass the \`Button\` component \`size={size}\` 1`] = `
<Button
  hasShadow={false}
  icon={null}
  isCompact={false}
  isDisabled={false}
  isFluid={false}
  isFormSubmit={false}
  isLoading={false}
  isOutlined={false}
  isPositionedRight={false}
  isRounded={false}
  isSecondary={false}
  onClick={[Function]}
  size="massive"
>
  <Button
    as="button"
    basic={false}
    circular={false}
    className=""
    compact={false}
    disabled={false}
    floated="left"
    fluid={false}
    loading={false}
    onClick={[Function]}
    secondary={false}
    size="massive"
    type="button"
  >
    <button
      className="ui massive left floated button"
      onClick={[Function]}
      role="button"
      type="button"
    >
      Press me
    </button>
  </Button>
</Button>
`;

exports[`<Button /> should render a the right structure 1`] = `
<Button
  hasShadow={false}
  icon={null}
  isCompact={false}
  isDisabled={false}
  isFluid={false}
  isFormSubmit={false}
  isLoading={false}
  isOutlined={false}
  isPositionedRight={false}
  isRounded={false}
  isSecondary={false}
  onClick={[Function]}
  size={null}
>
  <Button
    as="button"
    basic={false}
    circular={false}
    className=""
    compact={false}
    disabled={false}
    floated="left"
    fluid={false}
    loading={false}
    onClick={[Function]}
    secondary={false}
    size={null}
    type="button"
  >
    <button
      className="ui left floated button"
      onClick={[Function]}
      role="button"
      type="button"
    >
      Press me
    </button>
  </Button>
</Button>
`;
